src/cli/index.js

```diff
@@ -90,133 +94,195 @@ try {
   process.exit(1);
 }
 
-const questions = [
-  {
-    type: 'list',
-    pageSize: 10,
-    name: 'template',
-    message: 'InstantSearch template',
-    choices: () => {
-      const templatesByCategory = getTemplatesByCategory();
-
-      return Object.entries(templatesByCategory).reduce(
-        (templates, [category, values]) => [
-          ...templates,
-          new inquirer.Separator(category),
-          ...values,
-        ],
-        []
-      );
-    },
-    validate(input) {
-      return Boolean(input);
+const questions = {
+  application: [
+    {
+      type: 'list',
+      pageSize: 10,
+      name: 'template',
+      message: 'InstantSearch template',
```

Comment:
if this question comes first, we can make only the questions after this conditional, and no longer need the new "implementation type" question